ekglib/ldap_parser/parser.py

```diff
@@ -500,23 +506,27 @@ def _parse_attribute(self, key, values):
         elif key == 'entryUUID':
             self._parse_entry_uuid(values)
         elif key == 'jpegPhoto':
-
-            log(f"found jpegPhoto {values}")
-            try:
-                for v in values:
-                    b = bytes(v, 'utf-8')
-                    base64_bytes = base64.b64encode(b)
-                    log(f"found jpegPhoto {base64_bytes}")
-                    base64_url = "data:image/jpeg;charset=utf-8;base64," + base64_bytes
-                    log(f"found jpegPhoto {base64_url}")
-                    self._add((self.entry_iri, LDAP.term(key), base64_url))
-            except:
-                log_error("problem reading jpegPhoto")
-            return
+            self.parse_binary_content(key, values)
         else:
             for value in values:
                 self._add((self.entry_iri, LDAP.term(key), Literal(value)))
 
+    def parse_binary_content(self, key, values):
+        """Binary content is transformed into a Literal with base64 encoding"""
+        for value in values:
+            base64_bytes = b64encode(str_to_binary(value)).decode('ascii')
```

Comment:
ideally, we should even have str at the point were here. right?

Reply:
yes, value is a str, but we need to extract the lower 8 bits of the unicode character, because the LDAP library returns everything as strings. Using utf-8 won't help, as it adds extra bytes to the stream

Reply:
aren't we using this: https://www.python-ldap.org/en/python-ldap-3.3.0/bytes_mode.html
suggests it could/should be returning bytes for EVERYTHING unless we've set bytes_mode=True (backwards compatible):

Reply:
Good find, I can play with that and see if it helps!

Reply:
The parser uses https://github.com/cannatag/ldap3 library, not python-ldap. Do we want the hassle to rewrite this right now? And my "workarounds" should work for now :-)